osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -59,7 +59,8 @@ public Task<multiplayer_room> GetRoomAsync(long roomId)
 
         public Task<multiplayer_playlist_item> GetCurrentPlaylistItemAsync(long roomId)
         {
-            return connection.QuerySingleAsync<multiplayer_playlist_item>("SELECT * FROM multiplayer_playlist_items WHERE room_id = @RoomID", new
+            // Todo: Add ordering.
+            return connection.QueryFirstAsync<multiplayer_playlist_item>("SELECT * FROM multiplayer_playlist_items WHERE room_id = @RoomId AND expired = 0", new
```

Comment:
Not sure about the `Single` -> `First` change. Seems like keeping `Single` would help in checking the invariant that there can be only one non-expired playlist item at a time.

Reply:
This is not going to be the case in the future. I envisioned two extra modes for the multiplayer server:
1. Host rotate - players will be able to queue up maps and the server will automatically pick them based on something like
`SELECT * FROM multiplayer_playlist_items WHERE room_id = @RoomId AND expired = 0 ORDER BY playlist_order ASC` instead of this line here.
2. Tournament - the server sets up a playlist where the user can change to any given item from that playlist but not add additional items to it. Again this line will change.

More refactoring is required before that path though.

Reply:
This is indeed the correct path forward. We want to be able to queue future playlist items, either as the host or via other systems.